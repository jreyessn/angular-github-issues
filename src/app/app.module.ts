import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppComponent } from './app.component';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { RouterModule } from '@angular/router';
import { appRoutes } from './app.routing';
import { LayoutModule } from './layouts/layout.module';

@NgModule({
  declarations: [
    AppComponent,
  ],
  imports: [
    BrowserModule,
    RouterModule.forRoot(appRoutes),
    BrowserAnimationsModule,

    // Layout module of your application
    LayoutModule,
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
