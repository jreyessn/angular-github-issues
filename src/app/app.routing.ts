import { Routes } from '@angular/router';
import { LayoutComponent } from './layouts/layout.component';

export const appRoutes: Routes = [

      // Redirect empty path to '/issues'
      { path: '', pathMatch : 'full', redirectTo: 'issues' },

      // Landing routes
      {
        path: '',
        component: LayoutComponent,
        children: [
          {
            path: 'issues',
            loadChildren: () => import('./issues/issues.module').then(m => m.IssuesModule)
          }
        ]
      }

];