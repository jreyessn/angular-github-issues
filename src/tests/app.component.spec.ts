import { TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { AppComponent } from '../app/app.component';

describe('AppComponent', () => {
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        RouterTestingModule
      ],
      declarations: [
        AppComponent
      ],
    }).compileComponents();
  });

  it('Debe crear el componente', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const component = fixture.componentInstance;
    expect(component).toBeTruthy();
  });

  it('Debe hacer el match con el snapshot', () => {
    const fixture  = TestBed.createComponent(AppComponent);
    const compiled = fixture.nativeElement as HTMLElement;
    fixture.detectChanges();
    expect(compiled).toMatchSnapshot()
  });



});
