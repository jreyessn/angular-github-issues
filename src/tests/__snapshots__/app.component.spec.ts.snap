// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent Debe hacer el match con el snapshot 1`] = `
<div
  id="root1"
>
  <router-outlet />
  
</div>
`;
